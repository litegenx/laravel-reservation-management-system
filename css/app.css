.router-link-active {
  cursor: pointer;
}

