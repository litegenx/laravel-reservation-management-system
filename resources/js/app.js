require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
