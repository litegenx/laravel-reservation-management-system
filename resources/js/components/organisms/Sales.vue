<template>
    <v-card
        raised
        class="pa-6 ma-4"
    >
        <v-card-title>
            {{ label }}
        </v-card-title>
        <v-card-text>
            <Bar
                :label="label"
                :dataset="data"
            />
            <v-overlay
                :value="isLoading"
                absolute
                class="text-center"
            >
                <v-progress-circular
                    indeterminate
                    size="32"
                />
            </v-overlay>
        </v-card-text>
    </v-card>
</template>

<script>
    import Bar from '../molecules/Bar';
    import { apiGet } from '../../utils/api';
    import { arrayToObject } from '../../utils/misc';

    export default {
        components: {
            Bar,
        },
        props: {
            label: {
                type: String,
                required: true,
            },
            start: {
                type: String,
                required: true,
            },
            end: {
                type: String,
                required: true,
            },
            type: {
                type: String,
                required: true,
            },
            roomId: {
                type: [Number, String],
                required: true,
            },
        },
        data () {
            return {
                data: [],
                isLoading: false,
            };
        },
        watch: {
            roomId: {
                handler: 'setup',
                immediate: true,
            },
        },
        methods: {
            async setup () {
                this.isLoading = true;
                const { response } = await apiGet('summary', {
                    data: {
                        'start_date': this.start,
                        'end_date': this.end,
                        type: this.type,
                        'room_id': this.roomId,
                    },
                });
                this.isLoading = false;
                if (response) {
                    this.data = Object.values(arrayToObject(Object.keys(response.data), {
                        getItem: key => ({ label: key, value: response.data[ key ] }),
                    }));
                }
            },
        },
    };
</script>
