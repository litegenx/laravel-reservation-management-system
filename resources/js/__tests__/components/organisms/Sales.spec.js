import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import Sales from '../../../components/organisms/Sales';
import app from '../app';
import setupLocalVue from '../setupLocalVue';

jest.useFakeTimers();
jest.mock('../../../utils/api', () => ({
    apiGet: () => ({}),
}));
jest.mock('vue-chartjs', () => ({
    Bar: {
        render: () => undefined,
        methods: {
            renderChart: () => undefined,
        },
    },
}));

describe('Sales', () => {
    it('should render sales', () => {
        const wrapper = mount(app(Sales, 'inner', {
            props: {
                label: 'test',
                start: '2020-01-01',
                end: '2020-01-31',
                type: 'test',
                roomId: 1,
            },
        }), setupLocalVue({
            store: new Vuex.Store({
                state: {
                    data: [],
                    isLoading: false,
                },
                getters: {
                    'loading/isLoading': () => true,
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.findAll('circle')).toHaveLength(1);
    });
});
