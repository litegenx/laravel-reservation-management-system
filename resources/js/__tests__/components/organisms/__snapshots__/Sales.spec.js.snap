// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sales should render sales 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          class="pa-6 ma-4 v-card v-card--raised v-sheet theme--light"
        >
          <div
            class="v-card__title"
          >
            
        test
    
          </div>
           
          <div
            class="v-card__text"
          >
            <!---->
             
            <div
              class="v-overlay text-center v-overlay--absolute v-overlay--active theme--dark"
              style="z-index: 5;"
            >
              <div
                class="v-overlay__scrim"
                style="opacity: 0.46; background-color: rgb(33, 33, 33); border-color: #212121;"
              />
              <div
                class="v-overlay__content"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-circular v-progress-circular--indeterminate"
                  role="progressbar"
                  style="height: 32px; width: 32px;"
                >
                  <svg
                    style="transform: rotate(0deg);"
                    viewBox="22.857142857142858 22.857142857142858 45.714285714285715 45.714285714285715"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      class="v-progress-circular__overlay"
                      cx="45.714285714285715"
                      cy="45.714285714285715"
                      fill="transparent"
                      r="20"
                      stroke-dasharray="125.664"
                      stroke-dashoffset="125.66370614359172px"
                      stroke-width="5.714285714285714"
                    />
                  </svg>
                  <div
                    class="v-progress-circular__info"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;
